lib/java/collections.nit

```diff
@@ -91,13 +133,47 @@ extern class JavaArray in "Java" `{ java.lang.Object[] `}
 	redef fun length in "Java" `{ return recv.length; `}
 end
 
-# TODO other primitive arrays:
-# * Java primitive array `byte[]`
-# * Java primitive array `short[]`
-# * Java primitive array `int[]`
-# * Java primitive array `long[]`
-# * Java primitive array `boolean[]`
-# * Java primitive array `char[]`
+# Java primitive array `boolean[]`
+extern class JavaBoolArray in "Java" `{ boolean[] `}
+	super AbstractJavaArray[Bool]
+
+	# Get a new array of the given `size`
+	new (size: Int) in "Java" `{ return new boolean[(int)size]; `}
+
+	redef fun [](i) in "Java" `{ return (boolean)recv[(int)i]; `}
```

Comment:
The `(boolean)` cast should be superfluous here, a Nit `Bool` is already a Java `boolean`.